pkg/kp/rollstore/consul_store.go

```diff
@@ -204,6 +218,15 @@ func (s consulStore) CreateRollingUpdateFromExistingRCs(u roll_fields.Update) (r
 		return roll_fields.Update{}, err
 	}
 
+	err = s.labelNewRC(u.NewRC, newRCLabels)
+	if err != nil {
+		// This is potentially bad, because the labels that were there
+		// before have been overwritten. However its unreasonable to
+		// expect that a rollback will succeed if the roll-forward
+		// didn't succeed, so just report the error
```

Comment:
In practice this doesn't pose a problem in Square's use of this, because no ongoing action is taken as a result of these lingering labels. Informational endpoints may have incorrect data, however